overlord/overlord.go

```diff
@@ -261,6 +265,18 @@ func (o *Overlord) SetRestartHandler(handleRestart func(t state.RestartType)) {
 func (o *Overlord) Loop() {
 	o.ensureTimerSetup()
 	o.loopTomb.Go(func() error {
+		if o.udevMon != nil {
+			err := o.udevMon.Connect()
```

Comment:
This would mean that when udev monitor Connect() or Run() fail we still run but hotplug no longer works? Perhaps we should take down the whole snapd process and let systemd restart it?

Reply:
Yes. I'm not sure, I wanted to be extra careful and here and see how it behaves. I'm open to discuss/change that.

Reply:
Ok, as discussed in the standup, I'll make it fail hard so that we can spot problems faster. For stable release we might consider more relaxed approach if needs be.

Reply:
Hmm, looking more into it, overlord's Loop() was not meant to propagate errors. If we somehow change this to die here, systemd will restart the job (and if udev support failed before, it will most likely fail again). So I think it's best to log the error and carry on, like we do for some internal errors as well. And I will add a simple spread test that ensures we don't hit this no our setups. How does that sound?

Reply:
It doesn't sound good. This error will basically say "hey, hotplug will never work again apparently", which is not a behavior we can accept. We also don't want to just restart snapd either, as there's a chance whatever caused the error is still there, which means snapd will keep restarting forever, so also unacceptable.

We need a different strategy which is resilient instead.